packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -73,6 +76,19 @@ export default class HttpMetaMaskDataAccess extends HttpDataAccess {
     channelId: string,
     topics?: string[],
   ): Promise<DataAccessTypes.IReturnPersistTransaction> {
+    const eventEmitter = new EventEmitter() as DataAccessTypes.PersistTransactionEmitter;
+
+    if (!this.persist) {
+      const result: DataAccessTypes.IReturnPersistTransaction = Object.assign(
+        eventEmitter,
+        getNoPersistTransactionRawData(topics),
+      );
+
+      // Emit confirmation instantly since data is not going to be persisted
+      result.emit('confirmed', result);
+      return result;
```

Comment:
This snippet appears 3x, isn't there a better pattern?

Reply:
Yes, I'm going to apply @alexandre-abrioux suggestion. I'll ask for a re-review.